internal/pkg/cli/deploy/workload.go

```diff
@@ -486,6 +491,26 @@ func (d *workloadDeployer) pushAddonsTemplateToS3Bucket() (string, error) {
 }
 
 func (d *workloadDeployer) runtimeConfig(in *StackRuntimeConfiguration) (*stack.RuntimeConfig, error) {
+	out, err := d.convertRuntimeConfig(in)
+	if err != nil {
+		return nil, err
+	}
+	// Assign defaults if not set.
+	if len(out.CustomResourcesURL) == 0 {
+		// Populate the custom resource paths that copilot should upload the resources to.
+		crs, err := d.customResources(d.templateFS)
+		if err != nil {
+			return nil, err
+		}
+		out.CustomResourcesURL = make(map[string]string)
+		for _, cr := range crs {
+			out.CustomResourcesURL[cr.Name()] = s3.URL(d.resources.Region, d.resources.S3Bucket, cr.ArtifactPath())
```

Comment:
What are your thoughts on handling all these in logic in `Template()` of `stack` package?

Reply:
That's a great idea, now I need to somehow find the time to do that refactor :rip: